lld/test/ELF/riscv-attributes.s

```diff
@@ -322,6 +331,49 @@
 #--- atomic_abi_A7.s
 .attribute atomic_abi, 3
 
+
+#      UNKNOWN_UNKNOWN: BuildAttributes {
```

Comment:
My other two comments were minor nits, this is the one where I'm wondering if there's some missing logic.

For every other case where one of the input files has the atomic_abi attribute and we produce an output, the output file also has the attribute. In this case, although one of the inputs had the attribute we end up dropping it in the output. I think that's probably not desired, but maybe I'm missing a reason to do this?

Reply:
So, we probably need to double check w/ @MaskRay, but I believe we don't emit attributes w/ 0 value in general. I think the following snippet seems to indicate that from a few lines below the call to `mergeAtomics`. I assume that is because the default value of 0 is pretty much 'not used', so it isn't emitted normally, and since UNKOWN is 0, it just falls into that case.

```
size_t size = 5 + merged.vendor.size() + 1 + 5;
  for (auto &attr : merged.intAttr)
    if (attr.second != 0)
      size += getULEB128Size(attr.first) + getULEB128Size(attr.second);
```

I'll update the patch to address the nits, but I'll try to wait for Fangrui to confirm if thats correct about 0 valued attributes before proceeding.

Reply:
The main reason it _might_ be helpful to differentiate between the attribute being present but UNKNOWN and the attribute not being present at all, is that if the attribute is there (and emitted when merging UNKNOWN with NONE) you at least know it's been processed by tools that were aware of the attribute. Is that useful? Not sure. It's probably not hugely important either way, but thought I'd check if this was by design or not.